fpdf/fpdf.py

```diff
@@ -867,6 +868,13 @@ def add_page(
             new_page=not self._has_next_page(),
         )
 
+        if self.background:
+            if isinstance(self.background, tuple):
+                self.set_fill_color(*self.background)
```

Comment:
Make sure to reset the fill color to the previous value afterwards, or any other filled elements added later may also get the background color.

Reply:
Done 👍 

I have a question regarding this though, if you don't mind
The reason I didn't restore the color was because the color was supposed to be restored right below that anyway. That's why I placed it right after the new page gets generated. At that point the old color is saved to the variable fc, and after the rectangle is drawn, the previous colors, font, etc are restored.

I've tested it before and there was no problem, the fill color I set right at the start didn't get overridden at any point and the small rectangles I drew as a test on multiple pages had the correct fill color.

What I didn't test before I saw your comment was what happens if no fill color is set. And in that case, the default fill color seems to not get restored. It does however get restored on a page without the color background without any further changes to it

I can't make sense of why other colors(including when the black color is actively set as a fill color) get restored but the default color doesn't.
Can you shed some light on this as I don't understand the underlying mechanism?

Reply:
I didn't look at the surrounding code when I wrote this, so take it just as a prophylactic comment of the style "please make sure that...".

Yes, `add_page()` resets all kinds of values, including colors  ([line 897ff](https://github.com/PyFPDF/fpdf2/blob/f6f6b0a27852bcca58263defdceaa4a946a5a4c9/fpdf/fpdf.py#L897)). So if you have verified that your change doesn't cause any havoc in that regard, you're fine.

Reply:
It actually did cause havoc in the end if the user doesn't set a fill color at any point, which is the part that's confusing me. It only causes issues with the default fill color. I'd expect them to set a fill color before filling something, but you can never know after all. So I did change it